### PR TITLE
[RL] enable multislice for validate_converter

### PR DESCRIPTION
**Pull request by khatwanimohit**
# Description

This PR makes changes to validate_converter.py so it can run on multislice environment which is helpful in validating the conversion for larger models. It also adds helper functions to compare golden_vllm_state and maxtext_vllm_state for layer-0 keys which is hidden behind a flag `debug_converter` and an option to upload layer-0 weights to GCS for finding the mismatch. 
It also uses tunix's reshard function for transferring params instead of a device_put

If the change fixes a bug or a Github issue, please include a link, e.g.,:
FIXES: b/123456
FIXES: #123456

*Notice 1:* Once all tests pass, the "pull ready" label will automatically be assigned.
This label is used for administrative purposes. Please do not add it manually.

*Notice 2:* For external contributions, our settings currently require an approval from a MaxText maintainer to trigger CI tests.

# Tests

Single host v5p run with Qwen3-30B: https://paste.googleplex.com/5229816551964672
tpu7x-128 run with Qwen3-30B with 64 devices for maxtext model and the rest for vllm model initialization: https://cloudlogging.app.goo.gl/evFutCTxYt9zC99j6

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/configs/base.yml [hunk 1]
```diff
@@ -1246,3 +1246,8 @@ elastic_enabled: false
 elastic_timeout_seconds: 300
 elastic_max_retries: 10
 elastic_min_slice_count: -1
+
+##### MaxText to VLLM Converter validation parameters
+vllm_load_format: dummy # Format to load the model for conversion. Options are "auto", "dummy"
+debug_converter: False # If True, run key coverage check, weight stats, and GCS upload then exit without generation.
+gcs_debug_path: "" # If set and debug_converter=True, upload converted layer-0 and global tensors as .npy to this GCS prefix.
```
Comment (hengtaoguo) [inline src/maxtext/configs/base.yml]: Since these three flags have been added to types.py already, is it still necessary to add in the base.yml?

File: src/maxtext/integration/vllm/torchax_converter/validate_converter.py [hunk 2]
```diff
@@ -90,32 +106,175 @@ def _clean_device_memory():
...
+      f"max={float(jnp.max(a)):.6f}"
   )
-  return model, mesh
 
 
-def save_dict_to_file(state_dict, filename):
-  with open(filename, "w", encoding="utf-8") as f:
-    for key in sorted(state_dict.keys()):
-      f.write(f"{key}: {state_dict[key].shape}\n")
+def _log_weight_stats(converted_state: dict, vllm_state: dict, compare: bool) -> None:
```
Comment (hengtaoguo) [inline src/maxtext/integration/vllm/torchax_converter/validate_converter.py:132]: I like this feature!

**Review by hengtaoguo — APPROVED** [review]
Thanks Mohit! After this upgrade, is it still runnable on a single TPU VM?

File: src/maxtext/configs/base.yml (hunk 1 above)
Comment (khatwanimohit) [reply src/maxtext/configs/base.yml]: I usually add cli args to both base.yml and types.py because I think not adding it in base.yml will not let me assign new values from CLI, I maybe wrong

File: src/maxtext/configs/base.yml [hunk 3]
```diff
@@ -1246,3 +1246,8 @@ elastic_enabled: false
 elastic_timeout_seconds: 300
 elastic_max_retries: 10
 elastic_min_slice_count: -1
+
+##### MaxText to VLLM Converter validation parameters
+vllm_load_format: dummy # Format to load the model for conversion. Options are "auto", "dummy"
```
Comment (xuefgu) [inline src/maxtext/configs/base.yml]: Should they be in base.yaml or rl.yaml?

File: src/maxtext/integration/vllm/torchax_converter/validate_converter.py [hunk 4]
```diff
@@ -80,6 +95,7 @@ def _setup_vllm_environment():
   os.environ["SKIP_JAX_PRECOMPILE"] = "1"
   os.environ["JAX_RANDOM_WEIGHTS"] = "False"
   os.environ["VLLM_ENABLE_V1_MULTIPROCESSING"] = "0"
+  os.environ["NEW_MODEL_DESIGN"] = "1"
```
Comment (xuefgu) [inline src/maxtext/integration/vllm/torchax_converter/validate_converter.py]: I would avoid hardcoding it in the application. The user who launches the command should set it explicitly.

File: src/maxtext/configs/base.yml (hunk 3 above)
Comment (khatwanimohit) [reply src/maxtext/configs/base.yml]: moved them to rl.yml

File: src/maxtext/integration/vllm/torchax_converter/validate_converter.py (hunk 4 above)
Comment (khatwanimohit) [reply src/maxtext/integration/vllm/torchax_converter/validate_converter.py]: removed

**Review by xuefgu — APPROVED** [review]

**Review by NicoGrande — APPROVED** [review]
